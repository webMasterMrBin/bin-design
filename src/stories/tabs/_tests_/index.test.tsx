import { render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Tabs, TabPane } from '../index';
import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('snapshot test', () => {
  it('basic use', () => {
    const wrapper = render(
      <Tabs defaultActiveKey={0}>
        <TabPane tab="tab1">TabContent1</TabPane>
        <TabPane tab="tab2">Tabcontent222</TabPane>
      </Tabs>
    );
    
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  
  it('add onChange', () => {
    const wrapper = render(
      <Tabs defaultActiveKey={0} onChange={() => console.log('now onchange')}>
        <TabPane tab="tab1">TabContent1</TabPane>
        <TabPane tab="tab2">Tabcontent222</TabPane>
      </Tabs>
    );
    
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
