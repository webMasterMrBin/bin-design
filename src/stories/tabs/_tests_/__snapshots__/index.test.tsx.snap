// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test add onChange 1`] = `
<div
  class="sc-bdVaJa glNQyH"
>
  <div
    class="tabs-header"
  >
    <div
      class="active tabPane-content"
    >
      tab1
    </div>
    <div
      class="tabPane-content"
    >
      tab2
    </div>
  </div>
  <div
    class="tabs-content"
  >
    tab1
  </div>
</div>
`;

exports[`snapshot test basic use 1`] = `
<div
  class="sc-bdVaJa glNQyH"
>
  <div
    class="tabs-header"
  >
    <div
      class="active tabPane-content"
    >
      tab1
    </div>
    <div
      class="tabPane-content"
    >
      tab2
    </div>
  </div>
  <div
    class="tabs-content"
  >
    tab1
  </div>
</div>
`;
